dags/inference/configs/jetstream_pytorch_gce_config.py

```diff
@@ -0,0 +1,164 @@
+
+PROJECT_NAME = Project.CLOUD_ML_AUTO_SOLUTIONS.value
+RUNTIME_IMAGE = RuntimeVersion.TPU_UBUNTU2204_BASE.value
+GCS_SUBFOLDER_PREFIX = test_owner.Team.INFERENCE.value
+
+
+def get_jetstream_pytorch_inference_nightly_config(
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    test_mode: common.SetupMode,
+    project_name: str = PROJECT_NAME,
+    runtime_version: str = RUNTIME_IMAGE,
+    network: str = "default",
+    subnetwork: str = "default",
+    is_tpu_reserved: bool = True,
+    num_slices: int = 1,
+    model_configs: Dict = {},
+):
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.BENCHMARK_DATASET,
+  )
+
+  set_up_cmds = (
+      "pip install --upgrade pip",
+      # Download jetstream and maxtext
+      "git clone https://github.com/google/jetstream-pytorch.git",
+      # Create a python virtual environment
+      "sudo apt-get -y update",
+      "sudo apt-get -y install python3.10-venv",
+      "sudo apt-get -y install jq",
+      "python -m venv .env",
+      "source .env/bin/activate",
+      # Setup MaxText & JetStream
+      "cd jetstream-pytorch && source install_everything.sh",
+  )
+
+  additional_metadata_dict = {
+      "checkpoint": f"{model_configs['checkpoint']}",
+      "dataset": f"{model_configs['dataset']}",
+      "max_output_length": f"{model_configs['max_output_length']}",
+      "batch_size": f"{model_configs['batch_size']}",
+  }
+
+  run_model_cmds = (
+      # Start virtual environment
+      "source .env/bin/activate",
+      # Get commit hash of the maxtext and jetstream repos
+      f"export METADATA_DICT='{json.dumps(additional_metadata_dict)}'",
+      'cd jetstream-pytorch && export JETSTREAM_PYTORCH_COMMIT_HASH=$(git log -1 --format="%H") && cd ..',
+      'cd jetstream-pytorch/deps/JetStream && export JETSTREAM_COMMIT_HASH=$(git log -1 --format="%H") && cd ../../..',
+      'export METADATA_DICT=$(jq -c \'. + { "jetstream_pytorch_commit_hash": $newVal}\' --arg newVal ${JETSTREAM_PYTORCH_COMMIT_HASH} <<<"$METADATA_DICT")',
+      'export METADATA_DICT=$(jq -c \'. + { "jetstream_commit_hash": $newVal}\' --arg newVal ${JETSTREAM_COMMIT_HASH} <<<"$METADATA_DICT")',
+      ### Benchmark
+      "cd jetstream-pytorch",
+      # Configure flags
+      f"export MODEL_NAME={model_configs['model_name']}",
+      f"export SIZE={model_configs['size']}",
+      f"export BATCH_SIZE={model_configs['batch_size']}",
+      f"export MAX_CACHE_LEN={model_configs['max_cache_length']}",
+      f"export CKPT_PATH={model_configs['checkpoint']}",
+      f"export TOKENIZER_PATH=$(pwd)/ckpt_dir/{model_configs['tokenizer']}",
+      f"export SHARDING_CONFIG={model_configs['sharding_config']}",
+      "mkdir ckpt_dir", 
+      "gsutil cp -r ${CKPT_PATH}/* ckpt_dir/",
+      "mkdir output_ckpt_dir",
+      """python -m convert_checkpoints \
```

Comment:
Can we run with converted weights? 
@yeandy For maxtext automation, do you need to convert the weight for every run?